source/features/stop-redirecting-in-notification-bar.tsx

```diff
@@ -7,11 +7,16 @@ const hasNotificationBar = (): boolean =>
 	JSON.parse(sessionStorage.notification_shelf ?? '{}').pathname === location.pathname;
 
 function handleClick(event: delegate.Event<MouseEvent, HTMLButtonElement>): void {
-	// Also restores the attribute on successive non-alt clicks
-	event.delegateTarget.form!.toggleAttribute('data-redirect-to-inbox-on-submit', !event.altKey);
+	// Disable the redirect to the Notifications inbox if either:
+	// 1. The alt key was held down during click
+	// 2. The notification has been opened in a new tab
+	const redirectDisabled = event.altKey || sessionStorage.rghIsNewTab === 'true';
+	event.delegateTarget.form!.toggleAttribute('data-redirect-to-inbox-on-submit', !redirectDisabled);
 }
 
 async function init(): Promise<void> {
+	// If the history length is `1` on init, the notification has been opened in a new tab
```

Comment:
I think the variable name is self-explanatory, no comment needed.

Reply:
Hmm, disagree here... The name in the line below is super hard to parse and understand.


```js
sessionStorage.rghIsNewTab = history.length === 1;
```

Also - I would like to explain my intent, in case our assumptions are wrong and this leads to a bug in the future.

So I have updated the comment to add more background / motivation information: b935c605660ac00809930533ff15aea9b0ffe83b

Reply:
Although I would suggest keeping the comment, If for whatever reason this is not acceptable to have guiding comments in the codebase, then I can remove.

Reply:
The line says:

- we want to know whether the current tab is a new tab
- we expect to determine this by measuring `history.length`

The comment says the same exact thing, in different words.

Comments are supposed to explain why you do something, not what you're doing bit by bit (unless it's a super complex line — which isn't the case)